test/linalg/triangular.jl

```diff
@@ -134,14 +144,26 @@ for elty1 in (Float32, Float64, Complex64, Complex128, BigFloat, Int)
         # Unary operations
         @test full(-A1) == -full(A1)
 
-        # copy
-        B = similar(A1)
-        copy!(B,A1)
-        @test B == A1
-        B = similar(A1.')
-        copy!(B, A1.')
-        @test B == A1.'
-
+        # copy and copy! (test views as well, see issue #14317)
+        let vrange = 1:n-1, viewA1 = t1(sub(A1.data, vrange, vrange))
+            # copy
+            @test copy(A1) == copy(full(A1))
+            @test copy(viewA1) == copy(full(viewA1))
+            # copy!
+            B = similar(A1)
+            copy!(B, A1)
+            @test B == A1
+            B = similar(A1.')
+            copy!(B, A1.')
+            @test B == A1.'
+            B = similar(viewA1)
+            copy!(B, viewA1)
+            @test B == viewA1
+            B = similar(viewA1.')
+            copy!(B, viewA1.')
+            @test B == viewA1.'
+        end
+        
```

Comment:
Fixed (I think), thanks! (I just set git to warn on trailing whitespace in commits via core.whitespace = trailing-space and correct on patch via apply.whitespace = fix. If there are better / other ways to preemptively catch whitespace errors, please let me know!)